### PR TITLE
First draft of release process documentation

### PR DESCRIPTION
**Pull request by sxa555**
An initial document to describe the process of doign quartlery and point releases for both HotSpot and OpenJ9 at AdoptOpenJDK

Signed-off-by: Stewart Addison <sxa@uk.ibm.com>

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+### Oracle Managed OpenJDK Quarterly PSU Update
+
+- For the most recent JDK release, updates are managed by Oracle, and there will only be two of them, e.g. `jdk-13.0.1` and `jdk-13.0.2` after the initial GA unless it is an LTS release
+- Oracle work on the quarterly updates internally and their these branches are not public until GA date along with the Vulnerability Group fixes.
+- On GA day, Oracle merges the internal branch and any Vulnerability Group fixes to produce the final GA build, and tagged, e.g. `jdk-13.0.1+9`, `jdk-13.0.1-ga`
+
+### Eclipse OpenJ9/OMR releases
+
+- The OpenJ9 releases are based on three codebases
+  - https://github.com/eclipse/openj9-omr (a platform abstraction layer which OpenJ9 builds use, based on https://github.com/eclipse/openj9-omr)
```
Comment (pshipton) [inline RELEASING.md]: should be based on https://github.com/eclipse/omr

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  - https://github.com/eclipse/openj9 (The JVM code)
+  - The "extensions" repository for each release which contains a modified version of the OpenJDK codebase (OpenJDK updates are merged by the IBM team who owns the extensions repository, not the AdoptOpenJDK project)
```
Comment (pshipton) [inline RELEASING.md]: Useful to have an example link here, like https://github.com/ibmruntimes/openj9-openjdk-jdk which is the head stream (next version of openjdk)

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+
+# Extra OpenJ9 prerequisite steps (skip for a HotSpot release)
+
+1. The extensions release branch (e.g. `openj9-0.17.0`) will exist from doing the milestone builds (OpenJ9 milestone process is covered in a later section)
+2. Ask the extensions team to run their release-specific merge jobs to ensure they are up to date - this is not done by jobs at AdoptOpenJDK
+3. Once merged to master, ask the extensions team to merge to openj9-staging branch
+3. Having merged to openj9-staging successfully, then a job hosted by the extensions team (not at AdoptOpenJDK) will have automatically been triggered, which will perform "sanity" testing on all platforms for the new `openj9-staging` branch and if successful testing will automatically promote the code to the "openj9" branch, acceptance jobs: https://ci.eclipse.org/openj9/
+4. OpenJ9 leads (Currently Peter Shipton and Dan Heidinga) will now verify the Eclipse OpenJ9 and OMR release branch against the newly merged openjdk GA level in openj9-staging. If they are happy they will tag their release branches with the release tag, eg. "openj9-0.17.0", you are then ready to build the release.
```
Comment (pshipton) [inline RELEASING.md]: https://github.com/pshipton and https://github.com/danheidinga

File: RELEASING.md (hunk 2 above)
Comment (sxa555) [reply RELEASING.md]: I'll add something like `see next glossary entry` where I have provided an example link into it. Was trying not to dilute that section by talking about extensions just yet. I did consider mergin both sections though

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+     * Announce: https://mail.openjdk.java.net/pipermail/jdk-updates-dev/
+   * jdkNN: https://hg.openjdk.java.net/jdk/jdkNN/
+     * Announce: https://mail.openjdk.java.net/pipermail/jdk-dev/
+
+
+# Extra OpenJ9 prerequisite steps (skip for a HotSpot release)
+
+1. The extensions release branch (e.g. `openj9-0.17.0`) will exist from doing the milestone builds (OpenJ9 milestone process is covered in a later section)
+2. Ask the extensions team to run their release-specific merge jobs to ensure they are up to date - this is not done by jobs at AdoptOpenJDK
+3. Once merged to master, ask the extensions team to merge to openj9-staging branch
```
Comment (andrew-m-leonard) [inline RELEASING.md]: This is done automatically at the same time as merging into master, it is merged into openj9-staging

Comment (sxa555) [reply RELEASING.md]: Removed that step now

**Comment by sxa555** [comment]
@andrew-m-leonard Are you ok with this now?

**Comment by andrew-m-leonard** [comment]
Yes...
note we need this document to be easily changed, otherwise it will get out of date/wrong. The beauty of our internal Box doc is we can correct/change things in seconds, like we did yesterday... We need the same here, ie. just create a PR and merge, I don't want to go through a review process to move one word.....

**Comment by sxa555** [comment]
@andrew-m-leonard This repository does not mandate reviewing before merging - if such changes need to be pushed straight through they can be

**Pull request merged**